components/sup/src/manager/service_updater/package_update_worker.rs

```diff
@@ -60,38 +61,44 @@ impl From<&Service> for PackageUpdateWorker {
 }
 
 impl PackageUpdateWorker {
-    /// Use the specified package ident to search for packages.
+    /// Use the specified package ident to search for packages taking into account the update
+    /// condition.
     ///
     /// If a fully qualified package ident is used, the future will only resolve when that exact
     /// package is found.
     // TODO (DM): The returned package ident should use FullyQualifiedPackageIdent.
-    pub async fn update_to(&self, install_ident: PackageIdent) -> PackageIdent {
-        let install_source = install_ident.clone().into();
+    pub async fn update_to(&self, ident: PackageIdent, condition: UpdateCondition) -> PackageIdent {
         let delay = PackageUpdateWorkerPeriod::get();
         loop {
-            // TODO (DM): The entire behavior of the updater depends on this `install` function. It
-            // is what the old updater used, but its exact behavior should be tested/documented.
-            match util::pkg::install_no_ui(&self.builder_url, &install_source, &self.channel).await
-            {
+            let package_result = match condition {
+                UpdateCondition::Latest => {
+                    let install_source = ident.clone().into();
+                    util::pkg::install_no_ui(&self.builder_url, &install_source, &self.channel).await
+                }
+                UpdateCondition::TrackChannel => {
+                    util::pkg::install_channel_head(&self.builder_url, &ident, &self.channel).await
```

Comment:
I think you might be able to dispense with this `install_channel_head` function by taking advantage of the `LocalPackageUsage::Ignore` enum member.

(It still may be a good idea to have a separate function that hides the usage of that, to keep the code clear and intention-revealing, but that may keep the amount of extra code written to a minimum.)

Reply:
I tried this and it did not work. It looks like ignoring local packages only applies in [`determine_latest_from_ident`](https://github.com/habitat-sh/habitat/blob/b652179e772ddfd5a7ee324d4e5b24e4582ebbe1/components/common/src/command/package/install.rs#L463), but then right after that we check if the package is [installed](https://github.com/habitat-sh/habitat/blob/b652179e772ddfd5a7ee324d4e5b24e4582ebbe1/components/common/src/command/package/install.rs#L466) and use the local copy if it exists.

Maybe this is a bug and should be fixed?

Reply:
Hrmm... I don't think it's a bug, so much as it's an instance of the long-standing assumption that we always run the latest package on disk and never roll back coming into conflict with the goal of this PR :sweat_smile: 

I think there is probably a way to adjust the overall installation logic to cope with this, but that's likely a big lift (or at least, too big for this particular PR). Since the `LocalPackageUsage` flag is a bit of a work-around itself, this is probably a sign that such an adjustment is due. For the purposes of the current feature, though, I think the approach you've taken will be fine. 

Would you mind filing an issue to consolidate the installation logic and more holistically account for the local package usage as well as this rollback-related behavior?

Reply:
Issue #7531 for package installation consolidation.